connectors/sources/salesforce.py

```diff
@@ -0,0 +1,1453 @@
+                "Title",
+                "Body",
+                "LinkUrl",
+            ],
+        )
+        where_in_clause = ",".join(f"'{x}'" for x in case_ids)
+        join_clause = await self._case_feed_comments_join()
+
+        return (
+            SalesforceSoqlBuilder("CaseFeed")
+            .with_id()
+            .with_default_metafields()
+            .with_fields(queryable_fields)
+            .with_fields(["CreatedBy.Id", "CreatedBy.Name", "CreatedBy.Email"])
+            .with_join(join_clause)
+            .with_where(f"ParentId IN ({where_in_clause})")
+            .build()
+        )
+
+    async def _case_feed_comments_join(self):
+        queryable_fields = await self._select_queryable_fields(
+            "FeedComment",
+            [
+                "ParentId",
+                "CreatedDate",
+                "LastEditById",
+                "LastEditDate",
+                "CommentBody",
+                "IsDeleted",
+                "StatusParentId",
+            ],
+        )
+        return (
+            SalesforceSoqlBuilder("FeedComments")
+            .with_id()
+            .with_fields(queryable_fields)
+            .with_fields(["CreatedBy.Id", "CreatedBy.Name", "CreatedBy.Email"])
+            .with_limit(500)
+            .build()
+        )
+
+    async def content_document_links_join(self):
+        """Cached async property for getting downloadable attached files
+        This join is identical for all SObject queries"""
+        if self._content_document_links_join is not None:
+            return self._content_document_links_join
+
+        links_queryable = await self._is_queryable("ContentDocumentLink")
+        docs_queryable = await self._is_queryable("ContentDocument")
+        versions_queryable = await self._is_queryable("ContentVersion")
+        if not all([links_queryable, docs_queryable, versions_queryable]):
+            self._logger.warning(
+                "ContentDocuments, ContentVersions, or ContentDocumentLinks were not queryable, so not including in any queries."
+            )
+            self._content_document_links_join = ""
+            return self._content_document_links_join
+
+        queryable_docs_fields = await self._select_queryable_fields(
+            "ContentDocument",
+            [
+                "Title",
+                "FileExtension",
+                "ContentSize",
+                "Description",
+            ],
+        )
+        queryable_version_fields = await self._select_queryable_fields(
+            "ContentVersion",
+            [
+                "VersionDataUrl",
+                "VersionNumber",
+            ],
+        )
+        queryable_docs_fields = [f"ContentDocument.{x}" for x in queryable_docs_fields]
+        queryable_version_fields = [
+            f"ContentDocument.LatestPublishedVersion.{x}"
+            for x in queryable_version_fields
+        ]
+        [f"'{x[1:]}'" for x in TIKA_SUPPORTED_FILETYPES]
```

Comment:
Is this an unused statement? Not sure if it's a github PR formatting issue but I have an impression that we initialise this array without any reference to it.

Reply:
Ah that is also in the line below it, it's probably a reformatting error. Removed, thanks for spotting that.